### DISCUSSION TITLE
KnowledgeSystem/FrontEnd/Engineering/module

### DISCUSSION THREAD

**Discussion by giscus** [rockbenben/LearnData#46] [Comments]
# KnowledgeSystem/FrontEnd/Engineering/module

模块化是工程化的基础：只有能将代码模块化，拆分为合理单元，才具备调度整合的能力，才有架构和工程一说 早期，JavaScript 只是作为浏览器端脚本语言出现，只负责简单的页面交互，并不具备先天的模块化能力 随着 Node.js 的发展和 ES 的演进，模块化如今在前端领域早已经不新鲜，但是，对于模块化不应该只停留在了解、会用的基础上，还要深入其中，认识...

https://www.zxwin0125.top/KnowledgeSystem/FrontEnd/Engineering/module.html

**Comment by zxwin0125** [comment]
如果 module1 依赖外部模块 module2，该怎么办？
答：可以将 module2 作为一个参数传递给自调用的函数，这样可以确保 module2 在 module1 中是可用的
